pyth/tests/conftest.py

```diff
@@ -228,3 +230,63 @@ def pyth_add_product(solana_test_validator, pyth_dir, pyth_init_mapping):
     output = output.decode('ascii')
     output = output.splitlines()
     return output[0]
+
+
+@pytest.fixture(scope='session')
+def pyth_init_product(solana_test_validator, pyth_dir, pyth_add_product):
+
+    ltc = {
+        'account': pyth_add_product,
+        'attr_dict': {
+            'symbol': 'LTC/USD',
+            'asset_type': 'Crypto',
+            'country': 'US',
+            'quote_currency': 'USD',
+            'tenor': 'Spot',
+            'jlqd_symbol': 'LTCUSD',
+        },
+    }
+    fd, path = mkstemp(suffix='.json', prefix='ltc_')
+    with fdopen(fd, 'w') as f:
+        json.dump([ltc], f)
+    cmd = [
+        'pyth_admin', 'upd_product', path,
+        '-r', 'localhost',
+        '-k', pyth_dir,
+        '-c', 'finalized',
+    ]
+    check_call(cmd)
+    os.remove(path)
```

Comment:
This won't remove the temp file on an exception. I think it should be:
```
fd, path = mkstemp(suffix='.json', prefix='ltc_')
try:
    with fdopen(fd, 'w') as f:
        json.dump([ltc], f)
    cmd = [
        'pyth_admin', 'upd_product', path,
        '-r', 'localhost',
        '-k', pyth_dir,
        '-c', 'finalized',
    ]
    check_call(cmd)
finally:
    os.remove(path)
```

Reply:
that was intentional - if it fails we want to see the input

Reply:
Ok, but logging the file contents on exception before removing would probably be less annoying than remembering to manually remove the file after each failure.